src/qz/utils/PrintingUtilities.java

```diff
@@ -59,16 +82,7 @@ public static PrinterResolution getNativeDensity(PrintService service) {
         PrinterResolution pRes = (PrinterResolution)service.getDefaultAttributeValue(PrinterResolution.class);
 
         if (pRes == null && !SystemUtilities.isWindows()) {
-            if (CUPS_DPI == null || !CUPS_DESC.containsValue(service.getName())) {
-                CUPS_DESC = ShellUtilities.getCupsPrinters();
-                CUPS_DPI = ShellUtilities.getCupsDensities(CUPS_DESC);
-            }
-
-            for(String name : CUPS_DESC.keySet()) {
-                if (CUPS_DESC.get(name).equals(service.getName())) {
-                    return CUPS_DPI.get(name);
-                }
-            }
+            return CUPS_DPI.get(getPrinterId(service.getName()));
```

Comment:
Well the call to get `CUPS_DPI`'s values relies on `CUPS_DESC`, so I don't think we'd need to worry about a synchronization issue between what's available in each map. And if they are only raw printing, as this other use case for `CUPS_DESC` is, then they shouldn't have to worry about dpi anyway, right?